scrapy/utils/misc.py

```diff
@@ -160,6 +160,46 @@ def create_instance(objcls, settings, crawler, *args, **kwargs):
     return instance
 
 
+def build_from_crawler(objcls, crawler, /, *args, **kwargs):
+    """Construct a class instance using its ``from_crawler`` constructor.
+
+    ``*args`` and ``**kwargs`` are forwarded to the constructor.
+
+    Raises ``TypeError`` if the resulting instance is ``None``.
+    """
+    settings = crawler.settings
```

Comment:
As this is only used in the line 175, I'd just remove this var and used `crawler.settings` directly on that line.